macos/mac_os_resource_snd.ksy

```diff
@@ -0,0 +1,280 @@
+        enum: sound_header_type
+        doc: sample's encoding option
+      - id: base_frequency
+        type: u1
+        doc: base frequency of sample
+        -orig-id: baseFrequency
+      - id: num_frames
+        type: u4
+      - id: aiff_sample_rate
+        size: 10
+        doc: rate of original sample (Extended80)
+      - id: marker_chunk
+        -orig-id: markerChunk
+        type: u4
+        doc: reserved
+      - id: instrument_chunks
+        -orig-id: instrumentChunks
+        type: u4
+        doc: pointer to instrument info
+      - id: aes_recording
+        -orig-id: AESRecording
+        type: u4
+        doc: pointer to audio info
+      - id: sample_size
+        -orig-id: sampleSize
+        type: u2
+        doc: number of bits per sample
+      - id: reserved
+        -orig-id: futureUse1, futureUse2, futureUse3, futureUse4
+        size: 14
+        doc: reserved
+      - id: sample_area
+        -orig-id: sampleArea
+        size: num_frames * num_channels * sample_size / 8
+        doc: sampled-sound data
+  compressed_sound_header:
+    seq:
+      - id: sample_ptr
+        -orig-id: samplePtr
+        type: u4
+        doc: pointer to samples (or 0 if samples follow data structure)
+      - id: num_channels
+        -orig-id: numChannels
+        type: u4
+        doc: number of channels in sample
+      - id: sample_rate
+        -orig-id: sampleRate
+        type: unsigned_fixed_point
+        doc: The rate at which the sample was originally recorded.
+      - id: loop_start
+        -orig-id: loopStart
+        type: u4
+        doc: loop point beginning
+      - id: loop_end
+        -orig-id: loopEnd
+        type: u4
+        doc: loop point ending
+      - id: encode
+        type: u1
+        enum: sound_header_type
+        doc: sample's encoding option
+      - id: base_frequency
+        type: u1
+        doc: base frequency of sample
+        -orig-id: baseFrequency
+      - id: num_frames
+        type: u4
+      - id: aiff_sample_rate
+        size: 10
+        doc: rate of original sample (Extended80)
+      - id: marker_chunk
+        -orig-id: markerChunk
+        type: u4
+        doc: reserved
+      - id: format
+        size: 4
+        type: str
+        encoding: ASCII
+        doc: data format type
+      - id: reserved
+        -orig-id: futureUse2
+        size: 4
+        doc: reserved
+      - id: state_vars_ptr
+        -orig-id: stateVars
+        type: u4
+        doc: pointer to StateBlock
+      - id: left_over_samples_ptr
+        -orig-id: leftOverSamples
+        type: u4
+        doc: pointer to LeftOverBlock
+      - id: compression_id
+        -orig-id: compressionID
+        type: u2
+        doc: ID of compression algorithm
+      - id: packet_size
+        -orig-id: packetSize
+        type: u2
+        doc: number of bits per packet
+      - id: synthesizer_id
+        -orig-id: snthID
+        type: u2
+        doc: resource ID of Sound Manager Synthesizer (snth)
```

Comment:
I've found an explanation of this field in an older version of Inside Macintosh (Volume VI, dated 1991). A PDF copy of this version can be found [here](https://vintageapple.org/inside_o/pdf/Inside_Macintosh_Volume_VI_1991.pdf) (warning: it's 400 MB large!). The field is documented on page "22-49" (absolute page number 1169 in the PDF).

> <table><tbody><tr><td>snthID</td><td>Indicates the resource ID number of the 'snth' resource that was used to compress the packets contained in the compressed sound header. A 3:1 'snd ' resource would have a snthID of 11, and a 6:1 'snd ' would have a snthID of 13. If a compressed sound header contains samples that aren't compressed, you should set the snthID field to 0.</td></tr></tbody></table>

It seems that Apple stopped using this field, so later versions of the Sound Manager documentation just say that it's unused and should be set to 0. I think it would be good to point out that it's no longer used, but the previous meaning should still be documented.